data/scenarios/Challenges/wave.yaml

```
@@ -0,0 +1,74 @@
+version: 1
+name: Wave
+author: Karl Ostmo
+description: |
+  Ride the wave
+creative: false
+objectives:
+  - goal:
+      - |
+        Grab the `bitcoin`{=entity} at the east end of the path.
+        Don't let the patrolling robots catch you!
+    prerequisite:
+      not: got_caught
+    condition: |
+      as base {has "bitcoin"};
+  - id: got_caught
+    teaser: Got caught
+    optional: true
+    hidden: true
+    goal:
+      - |
+        The robots caught you!
+    condition: |
+      as base {x <- meet; return $ case x (\_. false) (\_. true)};
+robots:
+  - name: base
+    dir: [0, 1]
+    devices:
+      - branch predictor
+      - comparator
+      - dictionary
+      - grabber
+      - hourglass
+      - lambda
+      - logger
+      - net
+      - scanner
+      - strange loop
+      - treads
+  - name: wavebot
+    system: true
+    dir: [0, 1]
+    display:
+      invisible: false
+      attr: 'plant'
+    program: |
+      run "scenarios/Challenges/_wave/wavebot.sw"
```

This is a big performance problem, we need import (#495) to not pay for parsing 242 times.

```suggestion
      def crossPath =
          move; move; move; move; move; move;
          turn back;
          wait 5;
          end;
      def go =
          crossPath;
          go;
          end;
      def start =
          pos <- whereami;
          wait $ fst pos;
          go;
          end;
      start;
```